tests/dotnet/UnitTests/BundleStructureTest.cs

```diff
@@ -0,0 +1,525 @@
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+using NUnit.Framework;
+
+using Xamarin.Utils;
+
+namespace Xamarin.Tests {
+	[TestFixture]
+	public class BundleStructureTest : TestBaseClass {
+		void CheckAppBundleContents (ApplePlatform platform, string appPath, string[] runtimeIdentifiers, CodeSignature isSigned)
+		{
+			// Directory.GetFileSystemEntries will enter symlink directories and iterate inside :/
+			Console.WriteLine ($"App bundle: {appPath}");
+			Assert.AreEqual (0, ExecutionHelper.Execute ("find", new string [] { appPath }, out var output), "find");
+
+			var isCoreCLR = platform == ApplePlatform.MacOSX;
+			var allFiles = output.ToString ().
+								Split ('\n', StringSplitOptions.RemoveEmptyEntries).
+								Where (v => v.Length > appPath.Length).
+								Select (v => v.Substring (appPath.Length + 1)).ToList ();
+
+			// Remove various files we don't care about (for this test) from the list of files in the app bundle.
+			Predicate<string?> predicate = (v) => {
+				var fn = Path.GetFileName (v!);
+
+				switch (fn) {
```

Comment:
If/when the runtime adds more files to the bundle this test pat looks to break. Maybe add a comment here explaining to future us when to add more files to this list?